ui/client/components/graph/node-modal/editors/expression/EditableExpression.js

```diff
@@ -0,0 +1,76 @@
+import RawEditor from "./RawEditor"
+import BoolEditor from "./BoolEditor"
+import React from "react"
+import ExpressionWithFixedValues from "./ExpressionWithFixedValues"
+import {parseableBoolean} from "./ExpressionParser"
+
+export default class EditableExpression extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      displayRawEditor: true
+    }
+  }
+
+  render() {
+    const {fieldType, expressionObj, rowClassName, valueClassName, showSwitch} = this.props
+    const editorName = resolveEditorName(fieldType, expressionObj, this.state.displayRawEditor)
+    const Editor = editorTypes[editorName]
+    return <Editor toggleEditor={this.toggleEditor}
+                   switchable={this.switchable(editorName, expressionObj)}
+                   shouldShowSwitch={this.showSwitch(fieldType, showSwitch)}
+                   rowClassName={rowClassName ? rowClassName : "node-row"}
+                   valueClassName={valueClassName ? valueClassName : "node-value"}
+                   {...this.props}/>
+  }
+
+  toggleEditor = (_) => this.setState({
+    displayRawEditor: !this.state.displayRawEditor
+  })
+
+  switchable = (editorName, expressionObj) => {
+    switch (editorName) {
+      case boolEditor:
+        return true
+      case rawEditor:
+        return parseableBoolean(expressionObj)
+      default:
+        return false
+    }
+  }
+
+  showSwitch = (fieldType, showSwitch) =>
+    showSwitch &&
+    (!(fieldType === "VariableBuilder" || fieldType === "Variable")
```

Comment:
This strings should be defined as a const before componnet. Look at DeploymentAction in https://github.com/TouK/nussknacker/pull/518/files#diff-2bbdc336921dd7cd73e1f80fdff9f6b9R11

Reply:
done